runtime-modules/council/src/lib.rs

```diff
@@ -761,24 +782,33 @@ impl<T: Trait> EnsureChecks<T> {
     fn can_announce_candidacy(
         origin: T::Origin,
         council_user_id: &T::CouncilUserId,
+        staking_account_id: &T::AccountId,
         stake: &Balance<T>,
-    ) -> Result<(CouncilStageAnnouncing, T::AccountId), Error<T>> {
+    ) -> Result<CouncilStageAnnouncing, Error<T>> {
         // ensure user's membership
-        let account_id = Self::ensure_user_membership(origin, &council_user_id)?;
+        Self::ensure_user_membership_staking(origin, &council_user_id, staking_account_id)?;
 
         let stage_data = match Stage::<T>::get().stage {
             CouncilStage::Announcing(stage_data) => stage_data,
             _ => return Err(Error::CantCandidateNow),
         };
 
         // ensure that previous stake is unlockable (if any present)
-        Self::ensure_previous_stake_reusable(council_user_id, &account_id).map_err(|error| {
-            if error == Error::StakeStillNeeded {
-                return Error::CantCandidateTwice;
-            }
+        let old_candidate =
+            Self::ensure_previous_stake_is_reusable(council_user_id).map_err(|error| {
+                if error == Error::StakeStillNeeded {
+                    return Error::CantCandidateTwice;
+                }
+
+                error
+            })?;
 
-            error
-        })?;
+        // ensure existing and current staking account are the same
+        if let Some(tmp_old_candidate) = old_candidate {
```

Comment:
Why do these have to be the same? I should be able to stake with a new, or same, staking account associated with my same membership. 

Please explain.

What we do however have to do is to remove the lock that exists on that candidacy by virtue of failing to get elected from it _and_ not recovering the stake in the past. I am not 100% on whether my assumptions about cleanup are 100% correct, given that I am not on top of how you are using locks, but some variation of that should be true.

Reply:
In the current implementation, announcing candidacy can be done after releasing a previous candidacy stake is released with an exception when the user wants to reuse the stake (setting different stake amount is possible) -> that's why only the same account id is allowed.

We could change this, and announcing candidacy may release previous candidacy stake. Originally, I was afraid it could create some side effects, but after giving it a thought now, I think there is no reason to do so.

Reply:
Please do :)

Reply:
Done